packages/webcert/src/feature/careProvider/CareProviderModal.tsx

```diff
@@ -1,21 +1,50 @@
 import { PopUpModal, ResourceLinkType } from '@frontend/common'
 import _ from 'lodash'
-import React from 'react'
-import { useSelector } from 'react-redux'
-import { getUserResourceLinks, selectIsLoadingUserStatistics } from '../../store/user/userSelectors'
+import React, { useEffect } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import { updateIsCareProviderModalOpen } from '../../store/user/userActions'
+import { getUser, selectIsLoadingUserStatistics, getIsCareProviderModalOpen, getUserResourceLinks } from '../../store/user/userSelectors'
 import { CareProviderModalContent } from './CareProviderModalContent'
 
 const CareProviderModal: React.FC = () => {
+  const dispatch = useDispatch()
+  const user = useSelector(getUser)
   const isLoadingUserStatistics = useSelector(selectIsLoadingUserStatistics, _.isEqual)
+  const isCareProviderModalOpen = useSelector(getIsCareProviderModalOpen)
   const userLinks = useSelector(getUserResourceLinks)
   const chooseUnitLink = userLinks?.find((link) => link.type === ResourceLinkType.CHOOSE_UNIT)
+  const changeUnitLink = userLinks?.find((link) => link.type === ResourceLinkType.CHANGE_UNIT)
+
+  const getModalTitle = () => {
+    if (chooseUnitLink) {
+      return chooseUnitLink.name
+    } else if (changeUnitLink) {
+      return changeUnitLink.name
+    } else {
+      return ''
+    }
+  }
+
+  useEffect(() => {
+    if (chooseUnitLink && !user?.loggedInUnit.unitId) {
```

Comment:
Why is the logic to check if the unit id is not set necessary? Shouldn't it be enough with the resource link? If logical checks are needed these should be put in the backend with the resource link